### PR TITLE
Call to Action FSE Support

### PR DESCRIPTION
**Pull request by Gentyspun**
Ref: https://github.com/amnestywebsite/humanity-theme/issues/142

**Steps to test**:
1. Edit a post, the Call to Action block should now appear in the block inserter
2. Add a CTA block and populate with content
3. Save and view the front end.
4. CTA should display as expected

Example: http://bigbite.im/i/wR98Yf

### CODE REVIEW

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	 * @return string|null
+	 */
+	function amnesty_render_cta_block( $attrs, $content) {
+		// Apply filters to the attributes
+		$attrs = apply_filters( 'amnesty_cta_block_attributes', $attrs );
+
+		// Parse the attributes
+		$pre_heading = isset( $attrs['preheading'] ) ? $attrs['preheading'] : '';
+		$heading     = isset( $attrs['title'] ) ? $attrs['title'] : '';
+		$cta_content = isset( $attrs['content'] ) ? $attrs['content'] : '';
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:22]:
```suggestion
		$attrs = wp_parse_args(
			[
				'preheading' => '',
				'title'      => '',
				'content'    => '',
			],
			$attrs,
		);

		$pre_heading = $attrs['preheading'];
		$heading     = $attrs['title'];
		$cta_content = $attrs['content'];
```

**Review by jaymcp — CHANGES_REQUESTED** [review]
This needs a deprecation, so that old versions of the blocks can be transformed without breaking.

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php (hunk 1 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:22]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+if ( ! function_exists( 'amnesty_render_cta_block' ) ) {
+	/**
+	 * Render a download block
+	 *
+	 * @package Amnesty\Blocks
+	 *
+	 * @param array $attrs the block attributes
+	 *
+	 * @return string|null
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:13]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+				<p class="callToAction-content">%4$s</p>
+        		<div className="innerBlocksContainer">
+					%5$s
+        		</div>
+      		</div>',
+			esc_attr( $classes ),
+			wp_kses_post( $pre_heading ),
+			wp_kses_post( $heading ),
+			wp_kses_post( $cta_content ),
+			wp_kses_post( $content )
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+declare( strict_types = 1 );
+
+if ( ! function_exists( 'amnesty_render_cta_block' ) ) {
+	/**
+	 * Render a download block
+	 *
+	 * @package Amnesty\Blocks
+	 *
+	 * @param array  $attrs      the block attributes
+	 * @param string $content    the block content
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:12]:
```suggestion
	 * @param array  $attrs   the block attributes
	 * @param string $content the block content
```

**Review by jaymcp — DISMISSED** [review]
in `blocks/call-to-action/index.jsx`, the `classnames` import is no longer required, nor are `isEmpty` and `RichText`.

i'm not sure what's causing the issue, but the CTA doesn't render correctly with this change. 

1. i had an existing page containing CTA blocks, which converted fine, but have issues in the render due to a line break in the `aria-label` on the container. the background and demarcation lines are also missing:

2. i inserted new CTA blocks on a new page, and i only see the buttons being output:

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare( strict_types = 1 );
+
+if ( ! function_exists( 'amnesty_render_cta_block' ) ) {
+	/**
+	 * Render a download block
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:7]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php (hunk 4 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/call-to-action/render.php:12]: resolved

File: private/src/scripts/editor/blocks/call-to-action/deprecated.jsx [hunk 6]
```diff
@@ -26,6 +26,45 @@ const blockAttributes = {
   style: {
     type: 'string',
   },
+  actionType: {
+    type: 'string',
+  },
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply private/src/scripts/editor/blocks/call-to-action/deprecated.jsx:31]: Yep, resolved

**Review by jaymcp — APPROVED** [review]
Great! thanks for the PR, and for addressing the feedback!

**Pull request merged**